tcms/telemetry/static/telemetry/js/testing/status-matrix.js

```diff
@@ -0,0 +1,129 @@
+    };
+
+    document.getElementById('id_version').onchange = drawTable;
+    document.getElementById('id_build').onchange = drawTable;
+    document.getElementById('id_test_plan').onchange = drawTable;
+
+    $('#id_after').on('dp.change', drawTable);
+    $('#id_before').on('dp.change', drawTable);
+
+    drawTable();
+});
+
+
+function drawTable() {
+    if (table) {
+        table.destroy();
+    }
+
+    const query = {};
+
+    const productId = $('#id_product').val();
+    if (productId) {
+        query['product_version__product_id'] = productId;
+    }
+
+    const versionId = $('#id_version').val();
+    if (versionId) {
+        query['product_version_id'] = versionId;
+    }
+
+    const buildId = $('#id_build').val();
+    if (buildId) {
+        query['build_id'] = buildId;
+    }
+
+    const testPlanId = $('#id_test_plan').val();
+    if (testPlanId) {
+        query['plan_id'] = testPlanId;
+    }
+
+    const dateBefore = $('#id_before');
+    if (dateBefore.val()) {
+        query['stop_date__lte'] = dateBefore.data('DateTimePicker').date().format('YYYY-MM-DD 23:59:59');
+    }
+
+    const dateAfter = $('#id_after');
+    if (dateAfter.val()) {
+        query['start_date__gte'] = dateAfter.data('DateTimePicker').date().format('YYYY-MM-DD 00:00:00');
+    }
+
+    jsonRPC('Testing.status_matrix', query, data => {
+
+        const testRuns = [];
+
+        data.forEach(element => {
+            element.executions.forEach(testRun => {
+                testRuns[testRun.run_id] = testRun.run__summary;
+            });
+        });
+
+        for (let i = 1; i < testRuns.length; i++) {
+            if (testRuns[i]) {
+                d3.select('.table > thead > tr').append('th').text(`TR-${i}`);
+            }
+
+            table_columns.push({
+                data: null,
+                sortable: false,
+                render: function (data, type, full, meta) {
+                    const execution = full.executions.find(e => e.run_id === meta.col);
+                    if (execution) {
+                        return `<span class=${execution.status__name}>`
+                            + `<a href="/runs/${execution.run_id}/#caserun_${execution.pk}">TE-${execution.pk}</a>`
+                            + `</span>`;
+                    }
+                    return '';
+                }
+            });
+        }
+
+        table = $('#table').DataTable({
+            columns: table_columns,
+            data: data,
+            paging: false,
+            dom: "t"
+        });
+
+        table.cells()
+            .nodes()
+            .each(node => {
+                if (node.firstChild &&
+                    node.firstChild.classList &&
+                    node.firstChild.classList.length > 0) {
+
+                    const classList = [];
+                    node.firstChild.classList.forEach(className => classList.push(className.toLowerCase()));
+
+                    node.classList.add(classList);
+                }
```

Comment:
For each test execution, we draw a new cell, in this cell there is a `span` holding the test execution id and link to the test execution itself. 
```javascript
render: function (data, type, full, meta) {
                    const execution = full.executions.find(e => e.run_id === meta.col);
                    if (execution) {
                        return `<span class=${execution.status__name}>`
                            + `<a href="/runs/${execution.run_id}/#caserun_${execution.pk}">TE-${execution.pk}</a>`
                            + `</span>`;
                    }
                    return '';
                }
```
this `span` has applied style - the `status__name` of the execution that comes from the back-end. however, that style needs to be applied to the `td` element, not the `span` itself. with this block I am applying the style of the `span` to its parent element - the `td`. it may not be the most elegant solutions, but it's the only one that comes in mind